### PR TITLE
ci: Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by kyleam**
With the pending travis.org shutdown, we need to migrate to travis.com or another CI service.  This series switches to using GitHub actions.

All the old .travis.yml jobs should have a corresponding job in .github/workflows/test.yml.  As noted in the commit that adds the condor job, there was initially a condor_q-related core dump, but the latest runs haven't been triggering it, so (assuming it doesn't pop up again in this PR) that's obviously something to keep an eye on and ideally figure out.

### CODE REVIEW

**Comment by kyleam** [comment]
> assuming [the condor_q core dump] doesn't pop up again in this PR

It did:
https://github.com/ReproNim/reproman/pull/571/checks?check_run_id=1856173988

**Comment by yarikoptic** [comment]
reporting: I keep pestering @mih to give a try to new condor somehow ;)

**Comment by yarikoptic** [comment]
I have just uploaded 8.8.6~dfsg.1-1 to `/debian-devel` of neurodebian for where it built:
```
neurodebian@smaug ~/deb/builds/htcondor/8.8.6~dfsg.1-1 % grep -v OLD summary.build | grep OK
condor_8.8.6~dfsg.1-1~nd90+1_i386.build	OK	4:43.09 real, 278.53 user, 97.98 sys, 4980264 out
condor_8.8.6~dfsg.1-1~nd90+1_amd64.build	OK	5:01.60 real, 298.86 user, 130.28 sys, 5919456 out
condor_8.8.6~dfsg.1-1~nd100+1_i386.build	OK	6:51.86 real, 2159.03 user, 250.44 sys, 13854248 out
condor_8.8.6~dfsg.1-1~nd100+1_amd64.build	OK	6:58.10 real, 1974.24 user, 318.46 sys, 15727608 out
condor_8.8.6~dfsg.1-1~nd18.04+1_i386.build	OK	7:04.83 real, 2281.48 user, 247.67 sys, 11432416 out
condor_8.8.6~dfsg.1-1~nd18.04+1_amd64.build	OK	7:06.03 real, 2066.80 user, 321.39 sys, 12692800 out
```

note that 16.04 build failed, so we would need to do it on 18.04 where it seems to be ok

```
$ sed -e 's,/debian ,/debian-devel ,g' < /etc/apt/sources.list.d/neurodebian.sources.list | grep -v data > /etc/apt/sources.list.d/neurodebian-devel.sources.list

apt-get update

$ root@2235f71638d6:/# apt-cache policy htcondor
htcondor:
  Installed: (none)
  Candidate: 8.8.6~dfsg.1-1~nd18.04+1
  Version table:
     8.8.6~dfsg.1-1~nd18.04+1 500
        500 http://neuro.debian.net/debian-devel bionic/main amd64 Packages
     8.6.8~dfsg.1-2 500
        500 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
```

**Comment by kyleam** [comment]
> I have just uploaded 8.8.6~dfsg.1-1

Thanks!

**Comment by kyleam** [comment]
Hmm, it's at  ~40 minutes now, so it seems likely that something got stuck (perhaps `reproman run` waiting around because the newer version isn't giving the output it expects).  Will try to upgrade locally and debug.

**Comment by kyleam** [comment]
> so it seems likely that something got stuck

Based on the log [1], it's stuck waiting on a held job:

```
2021-02-09T21:29:31.1422549Z 2021-02-09 21:29:31,141 [INFO] Waiting on job 2: held. Next heartbeat in 766 seconds
```

<details>
<summary>debbugging in Ubuntu 18.04 VM</summary>

I installed condor 8.8.6 into a Ubuntu 18.04 VM.

Running from a new dataset in "/tmp/test", this simple job gets held:

```
reproman run -r local --follow --sub condor sh -c 'echo hey'
```

Here's the reason given by `condor_q --held`:

```
-- Schedd: kestrel : <127.0.0.1:9618?... @ 02/10/21 16:07:18
 ID      OWNER          HELD_SINCE  HOLD_REASON
   1.0   kyle            2/10 16:06 Error from slot1@kestrel: Failed to execute '/home/kyle/.reproman/run-root/20210210-160634-98b7/.reproman/jobs/local/20210210-160634-98b7/runscript' with arguments 0: Cannot access initial working directory "/tmp/test" (errno=2: 'No such file or directory')

Total for query: 1 jobs; 0 completed, 0 removed, 0 idle, 0 running, 1 held, 0 suspended
Total for kyle: 1 jobs; 0 completed, 0 removed, 0 idle, 0 running, 1 held, 0 suspended
Total for all users: 1 jobs; 0 completed, 0 removed, 0 idle, 0 running, 1 held, 0 suspended
```

So /tmp/test doesn't exist.  pytest would be operating from /tmp, so that's consistent with the CI getting stuck.

Here's how /tmp looks outside of the condor job:

```
drwxr-xr-x 5 kyle kyle 4096 Feb 10 16:04 /tmp/test
```

And here's what the condor job sees:

```sh
condor_version
condor_run 'ls -ld /tmp'
condor_run 'ls -l /tmp'
```

```
$CondorVersion: 8.8.6 Feb 09 2021 BuildID: Debian-8.8.6~dfsg.1-1~nd18.04+1 Debian-8.8.6~dfsg.1-1~nd18.04+1 $
$CondorPlatform: X86_64-Ubuntu_18.04 $
drwx------ 2 kyle kyle 4096 Feb 10 16:00 /tmp
total 0
```

Going back to the old, non-devel condor version (8.6.8), condor sees the regular /tmp directory, and /tmp/test _is_ there:

```
$CondorVersion: 8.6.8 Apr 06 2018 BuildID: Debian-8.6.8~dfsg.1-2 Debian-8.6.8~dfsg.1-2 $
$CondorPlatform: X86_64-Ubuntu_ $
drwxrwxrwt 16 root root 4096 Feb 10 16:02 /tmp
total 536
-rw------- 1 kyle kyle      0 Feb 10 13:55 config-err-PU1Ts6
drwx------ 2 kyle kyle   4096 Feb 10 13:55 ssh-7vProW84d1AH
drwx------ 3 root root   4096 Feb 10 13:55 systemd-private-b5c0ddacf32343c2b6c5ab1cc710c97c-bolt.service-HPDtLV
drwx------ 3 root root   4096 Feb 10 13:55 systemd-private-b5c0ddacf32343c2b6c5ab1cc710c97c-colord.service-UhRVFK
drwx------ 3 root root   4096 Feb 10 13:56 systemd-private-b5c0ddacf32343c2b6c5ab1cc710c97c-fwupd.service-Wtz62q
drwx------ 3 root root   4096 Feb 10 13:55 systemd-private-b5c0ddacf32343c2b6c5ab1cc710c97c-ModemManager.service-xwK4Y8
drwx------ 3 root root   4096 Feb 10 13:55 systemd-private-b5c0ddacf32343c2b6c5ab1cc710c97c-rtkit-daemon.service-sl9u5b
drwx------ 3 root root   4096 Feb 10 13:55 systemd-private-b5c0ddacf32343c2b6c5ab1cc710c97c-systemd-resolved.service-sqWSGj
drwx------ 3 root root   4096 Feb 10 13:55 systemd-private-b5c0ddacf32343c2b6c5ab1cc710c97c-systemd-timesyncd.service-JiTPUI
drwxr-xr-x 4 kyle kyle   4096 Feb 10 14:26 test
-rw------- 1 kyle kyle 511815 Feb 10 13:59 tmpaddon
```

</details>

[1] https://github.com/ReproNim/reproman/pull/571/checks?check_run_id=1866468976
    Sadly (but I think unrelatedly) those are filled by up many BlockingIOErrors).

**Comment by yarikoptic** [comment]
> [1] https://github.com/ReproNim/reproman/pull/571/checks?check_run_id=1866468976
> Sadly (but I think unrelatedly) those are filled by up many BlockingIOErrors).

oh no -- is that "from" datalad?

**Comment by kyleam** [comment]
> oh no -- is that "from" datalad?

Yes, I think so (based on not spotting them in builds without datalad)

**Comment by kyleam** [comment]
First non-stuck job with newer condor was green (2090aba).  I pushed another commit switching a py 3.6 job to 3.9, mostly to see whether it turns up green again (because with the older version it was flaky).  Hopefully that doesn't surface a 3.9-specfic error, though.

File: reproman/distributions/tests/test_venv.py [hunk 1]
```diff
@@ -62,8 +63,8 @@ def venv_test_dir():
     if ssp:
         # The testing environment supports --system_site_packages.
         pip2 = op.join("venv-nonlocal", "bin", "pip")
-        runner.run(["virtualenv", "--python", PY_VERSION,
-                    "--system-site-packages", "venv-nonlocal"])
+        runner.run(venv_cmd + ["--python", PY_VERSION,
```
Comment (kyleam) [inline reproman/distributions/tests/test_venv.py:66]: I'm confused by the lack of coverage here.  There is a dedicated job that sets should sets `REPROMAN_TESTS_ASSUME_SSP=1`, so this should be triggered.  I expected to have to do more to make the --system-site-packages job work (if even feasible), so it not being in effect would explain why it "just worked".

**Comment by kyleam** [comment]
There have been four non-stuck runs with condor 8.8, and none of them triggered the core dump.  With condor 8.6, it wasn't reliably triggered either, though there wasn't a streak of four runs without it.  So, not conclusive, but I think there's a good chance using 8.8 sidesteps the issue.

File: reproman/distributions/tests/test_debian.py [hunk 2]
```diff
@@ -77,7 +77,19 @@ def test_check_bin_packages():
...
+                assert getattr(o, a), "A non-local origin needs a " + a
+            except AssertionError:
+                # FIXME? On the Ubuntu GitHub CI run, this entry has an empty
+                # string for component.  Is that an issue?
+                #
+                # APTSource(name='apt_obs://build.opensuse.org/devel:kubic:...',
+                # component='', archive=None, architecture=None,
+                # codename='xUbuntu_18.04', ...)
+                if getattr(o, "label") == "devel:kubic:libcontainers:stable" \
+                   and a == "component" and "GITHUB_WORKFLOW" in os.environ:
```
Comment (yarikoptic) [inline reproman/distributions/tests/test_debian.py:90]: Didn't look in detail but I guess they simply provide a simplistic apt repo without proper structure/annotation. Given that continue is only for this one - looks good to me

**Review by yarikoptic — APPROVED** [review]
Thank you @kyleam , looks great to me, left one comment to agree on apt test workaround. Let's proceed

**Comment by yarikoptic** [comment]
And no immediate ideas about coverage ;-)

**Pull request merged**